### DISCUSSION TITLE
git@github.com: Permission denied (publickey).

### DISCUSSION THREAD

**Discussion by KristoferStoimenov** [community/community#47277] [New to GitHub]
Hello Guys, 
I get the current error 
```
git@github.com: Permission denied (publickey).
```
even after generating ssh keys on my local machine, and copying the public key to the settings of my github account.
What could the problem be?

**Comment by airtower-luna** [comment]
Try `ssh -vT git@github.com` to test the SSH connection itself and get debug info. Assuming you've added the public key to your account successfully, most likely the right key isn't tried for some reason.

**Comment by SMelidoni** [comment]
Hi @KristoferStoimenov 👋🏽

The "Permission denied (publickey)" error when attempting to use Git to connect to GitHub indicates that Git was unable to authenticate with your GitHub account using the SSH key that you generated.

Here are some steps you can take to troubleshoot this issue:

1. Check that you copied the correct public key to your GitHub account. To do this, navigate to your GitHub account settings and click on the "SSH and GPG keys" tab. Make sure the public key you added matches the one generated on your local machine.

2. Check that your local machine is using the correct private key when attempting to connect to GitHub. To do this, check the contents of your ~/.ssh/config file (if it exists). Make sure it specifies the correct private key file for the github.com host. For example:

```
Host github.com
  HostName github.com
  IdentityFile ~/.ssh/id_rsa_github
```

3. Make sure your local machine is trying to use the SSH protocol to connect to GitHub. To do this, check the remote URL for your Git repository. It should look like git@github.com:username/repo.git, with the git@github.com: prefix indicating that SSH is being used.

If none of these steps resolve the issue, you may want to try regenerating your SSH key and starting over.

Hope this helps!
